packages/core/src/new-api/internal/reconciliation/utils.ts

```diff
@@ -0,0 +1,81 @@
+import { IgnitionError } from "../../../errors";
+import { isRuntimeValue } from "../../type-guards";
+import {
+  AccountRuntimeValue,
+  Future,
+  ModuleParameterRuntimeValue,
+  ModuleParameterType,
+} from "../../types/module";
+import { isAddress } from "../utils";
+
+import { ReconciliationContext, ReconciliationFutureResult } from "./types";
+
+export function fail(
+  future: Future,
+  failure: string
+): ReconciliationFutureResult {
+  return {
+    success: false,
+    failure: {
+      futureId: future.id,
+      failure,
+    },
+  };
+}
+
+export function resolveFromAddress(
+  from: string | AccountRuntimeValue | undefined,
+  context: ReconciliationContext
+): string | undefined {
+  if (from === undefined || typeof from === "string") {
+    return from;
+  }
+
+  if (typeof from === "string") {
```

Comment:
This is never true because of the if in line 30

Reply:
Nice catch. I have removed the string comparison on line 30, so this check is now hit.